taui/src/selectors/est-max-rent.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import get from 'lodash/get'
+import {createSelector} from 'reselect'
+
+import detailNeighborhood from './detail-neighborhood'
+
+export default createSelector(
+  state => get(state, 'data.userProfile'),
+  detailNeighborhood,
+  (userProfile, neighborhood) => {
+    if (!userProfile || !neighborhood) {
+      return 0
+    }
+    return neighborhood && userProfile && neighborhood.properties['max_rent_' + userProfile.rooms + 'br']
```

Comment:
The guards on this line shouldn't be necessary because of the guard clause above that's returning 0, but they're also not going to protect against every case because `userProfile.rooms` and `neighborhood.properties.max_rent_1_br` could also be undefined. I'm not sure whether we have [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) available on this project, but that might be one way to protect this better. Otherwise I think adding guards for those other two properties should do the trick.

Reply:
Ah, yup, that makes sense. Looks like optional chaining isn't available (it's interpreting my `?` as a ternary operator) so I expanded the if statement above and hopefully that should cover it more effectively.